src/Connector.SqlServer/Utility/SanitizedSqlName.cs

```diff
@@ -0,0 +1,47 @@
+﻿using CluedIn.Connector.Common.Helpers;
+using System;
+
+namespace CluedIn.Connector.SqlServer.Utility
+{
+    public class SanitizedSqlName
+    {
+        protected readonly string Source;
+        protected string Sanitized;
+
+        /// <summary>
+        /// String sanitized for use in SQL code
+        /// </summary>
+        /// <param name="source"></param>
+        /// <exception cref="ArgumentNullException">Source string can't be null</exception>
+        public SanitizedSqlName(string source)
+        {
+            Source = source ?? throw new ArgumentNullException($"{nameof(source)} can't be null");
+        }
+
+        public virtual string GetValue()
+        {
+            if (Sanitized == null)
+                Sanitized = SqlStringSanitizer.Sanitize(Source);
+
+            return Sanitized;
+        }
+
+        public override string ToString()
+        {
+            return GetValue();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj == null || obj is SanitizedSqlName target == false)
+                return false;
+
+            return GetValue() == target.GetValue();
```

Comment:
how come this is `the use of unassigned variable _target_` ?
![image](https://user-images.githubusercontent.com/25682289/195837460-5368b1cb-0952-4677-8c88-ae8abac93afc.png)